kasa/smart/smartdevice.py

```diff
@@ -167,7 +167,8 @@ async def _initialize_modules(self):
                     mod.__name__,
                 )
                 module = mod(self, mod.REQUIRED_COMPONENT)
-                self.modules[module.name] = module
+                if await module._check_supported():
```

Comment:
We should probably do no I/O in this check (i.e., no await), if it's not required.

Reply:
I thought it would be useful to have the possibility of trying an api call if we need to in the future, i.e. for other cases like `get_device_usage` where we can catch an error once and then not try again.  Let me know if you want keep as sync though.

Reply:
Yes, this is fine and we can change it if need to be as it's our internal.